### DISCUSSION TITLE
Error: grid: http://node3:9000 connecting to ws://node1:9000/minio/grid/v1: unexpected HTTP response status: 403 (ws.StatusError) Sleeping 3.318s (3) (*fmt.wrapError)

### DISCUSSION THREAD

**Discussion by Misszhangruihua** [minio/minio#18832] [Q&A]
Error: grid: http://node3:9000 connecting to ws://node1:9000/minio/grid/v1: unexpected HTTP response status: 403 (ws.StatusError) Sleeping 3.318s (3) (*fmt.wrapError) 

I encountered this error while expanding the service pool. How can I resolve this error

**Comment by klauspost** [comment]
Most likely credentials are mismatching. Provide more information and we may be able to help more.

**Reply by Misszhangruihua** [reply]
I did not use TLS/SSL Certificates when configuring distributed mini, nor did I add them when expanding the service pool. Is this necessary

**Reply by klauspost** [reply]
The clusters must be uniform. Want better answers, provide all your setup information.

**Reply by Misszhangruihua** [reply]
/etc/default/minio

MINIO_VOLUMES="http://node{1...2}:9000/mnt/data-{1...4}"

MINIO_OPTS="--console-address :9001"

MINIO_ROOT_USER=minioadmin

MINIO_ROOT_PASSWORD=minioadmin

MINIO_SERVER_URL="http://node1:9000"


#####################   /usr/lib/systemd/system/minio.service


[Unit]
Description=MinIO
Documentation=https://min.io/docs/minio/linux/index.html
Wants=network-online.target
After=network-online.target
AssertFileIsExecutable=/opt/soft/minio

[Service]
WorkingDirectory=/opt/soft/

User=minio
Group=minio
ProtectProc=invisible


EnvironmentFile=-/etc/default/minio
ExecStartPre=/bin/bash -c "if [ -z \"${MINIO_VOLUMES}\" ]; then echo \"Variable MINIO_VOLUMES not set in /etc/default/minio\"; exit 1; fi"
ExecStart=/opt/soft/minio server $MINIO_OPTS $MINIO_VOLUMES


Restart=always

LimitNOFILE=65536

TasksMax=infinity

TimeoutStopSec=infinity
SendSIGKILL=no

[Install]
WantedBy=multi-user.target



The above is the original service pool configuration ，The expanded service configuration has only been modified  MINIO_VOLUMES="http://node3:9000/mnt/data-{1...16}"

**Reply by Misszhangruihua** [reply]
MINIO_VOLUMES="http://node{1...2}:9000/mnt/data-{1...4} http://node3:9000/mnt/data-{1...16}"

**Reply by Misszhangruihua** [reply]
1月 19 19:42:31 node3 minio[3827]: 8: internal/logger/logger.go:259:logger.LogIf()
1月 19 19:42:31 node3 minio[3827]: 7: cmd/prepare-storage.go:254:cmd.connectLoadInitFormats()
1月 19 19:42:31 node3 minio[3827]: 6: cmd/prepare-storage.go:312:cmd.waitForFormatErasure()
1月 19 19:42:31 node3 minio[3827]: 5: cmd/erasure-server-pool.go:120:cmd.newErasureServerPools()
1月 19 19:42:31 node3 minio[3827]: 4: cmd/server-main.go:1050:cmd.newObjectLayer()
1月 19 19:42:31 node3 minio[3827]: 3: cmd/server-main.go:790:cmd.serverMain.func10()
1月 19 19:42:31 node3 minio[3827]: 2: cmd/server-main.go:489:cmd.bootstrapTrace()
1月 19 19:42:31 node3 minio[3827]: 1: cmd/server-main.go:788:cmd.serverMain()
1月 19 19:42:31 node3 minio[3827]: Waiting for a minimum of 2 drives to come online (elapsed 20s)
1月 19 19:42:32 node3 minio[3827]: API: SYSTEM()
1月 19 19:42:32 node3 minio[3827]: Time: 11:42:32 UTC 01/19/2024
1月 19 19:42:32 node3 minio[3827]: Error: grid: http://node3:9000 connecting to ws://node1:9000/minio/grid/v1: EOF (*errors.errorString) Sleeping 2.94s (3) (*fmt.wrapError)
1月 19 19:42:32 node3 minio[3827]: 5: internal/logger/logonce.go:118:logger.(*logOnceType).logOnceIf()
1月 19 19:42:32 node3 minio[3827]: 4: internal/logger/logonce.go:149:logger.LogOnceIf()
1月 19 19:42:32 node3 minio[3827]: 3: internal/grid/connection.go:661:grid.(*Connection).connect.func1()
1月 19 19:42:32 node3 minio[3827]: 2: internal/grid/connection.go:690:grid.(*Connection).connect()
1月 19 19:42:32 node3 minio[3827]: 1: internal/grid/connection.go:245:grid.newConnection.func1()
1月 19 19:42:32 node3 minio[3827]: API: SYSTEM()
1月 19 19:42:32 node3 minio[3827]: Time: 11:42:32 UTC 01/19/2024
1月 19 19:42:32 node3 minio[3827]: Error: Read failed. Insufficient number of drives online (*errors.errorString)

**Reply by klauspost** [reply]
Are you running the same server versions on all machines? Which?

**Reply by klauspost** [reply]
Also, you must modify the *entire* cluster with the new configuration.

It is not enough just to add it to the new server.

**Reply by Misszhangruihua** [reply]
new node scp -r /opt/soft/minio node3:/opt/soft/

**Reply by klauspost** [reply]
?

**Reply by klauspost** [reply]
`MINIO_VOLUMES` must be the same on all servers.

**Reply by Misszhangruihua** [reply]
Does expanding the cluster require modifying all services to include the previous old cluster configuration? So this cluster needs to be restarted, can't the expansion provide external services

**Reply by Misszhangruihua** [reply]
Thank you for your answer. I'm giving it a try

**Reply by klauspost** [reply]
It does.

**Reply by Misszhangruihua** [reply]
This problem has been solved, thank you very much

**Reply by Misszhangruihua** [reply]
Can Minio dynamically add nodes? I originally thought this method was to dynamically add nodes, modify the original service pool configuration, and restart it to take effect.

**Reply by klauspost** [reply]
They can't just jump in and out, otherwise there could be no consistency in the cluster.
